tests/components/broadlink/test_config_flow.py

```diff
@@ -0,0 +1,748 @@
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], {"unlock": True},
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "unlock"
+    assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_flow_unlock_os_error(hass):
+    """Test we handle an OS error in the unlock step."""
+    device = get_device("Living Room")
+    mock_api = device.get_mock_api()
+    mock_api.is_locked = True
+    mock_api.set_lock.side_effect = OSError()
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("broadlink.discover", return_value=[mock_api]):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": device.host, "timeout": device.timeout},
+        )
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], {"unlock": True},
+    )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "unlock"
+    assert result["errors"] == {"base": "unknown"}
+
+
+async def test_flow_do_not_unlock(hass):
+    """Test we do not unlock the device if the user does not want to."""
+    device = get_device("Living Room")
+    mock_api = device.get_mock_api()
+    mock_api.is_locked = True
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    with patch("broadlink.discover", return_value=[mock_api]):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {"host": device.host, "timeout": device.timeout},
+        )
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], {"unlock": False},
+    )
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], {"name": device.name},
+    )
+
+    assert result["type"] == "create_entry"
+    assert result["title"] == device.name
+    assert result["data"] == device.get_entry_data()
+
+    assert mock_api.set_lock.call_count == 0
+
+
+async def test_flow_import_works(hass):
+    """Test an import flow."""
+    device = get_device("Living Room")
+    mock_api = device.get_mock_api()
+
+    with patch("broadlink.discover", return_value=[mock_api]) as mock_discover:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_IMPORT},
+            data={"host": device.host},
+        )
+
+    assert result["type"] == "form"
+    assert result["step_id"] == "finish"
+    assert result["errors"] == {}
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], {"name": device.name},
+    )
+
+    assert result["type"] == "create_entry"
+    assert result["title"] == device.name
+    assert result["data"]["host"] == device.host
+    assert result["data"]["mac"] == device.mac
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6042836)